newswires/client/src/context/SearchContext.tsx

```diff
@@ -110,9 +112,12 @@ export const SearchContext: Context<SearchContextShape | null> =
 	createContext<SearchContextShape | null>(null);
 
 export function SearchContextProvider({ children }: PropsWithChildren) {
-	const [currentConfig, setConfig] = useState<Config>(
+	const [currentConfig, setConfig] = useState<Config>(() =>
 		urlToConfig(window.location),
 	);
+	const [viewedItemIds, setViewedItemIds] = useState<string[]>(() =>
+		loadFromLocalStorage<string[]>('viewedItemIds', z.array(z.string()), []),
+	);
```

Comment:
as per your excellent segment in Engineering Meet-up, this will be run on every render, and I have a vague memory you'll get warnings if the value passed to initial state varies from render-to-render and even if not, it's probably wasteful, might be worth pulling out the call to `loadFromLocalStorage` into a `useMemo(    , [])` above the set state.

Reply:
Good point, but I think we're okay on this one, because the call to `loadFromLocalStorage` is wrapped in an anonymous function (albeit slightly obscured by the formatting here!) so will only get called once on initial render 👍